Test util.java and roster test update

navijation opened the pull request:
The TestUtil class is a utility class that can be used primarily to generate large test cases but also to accomplish other small, nonspecific tasks that may be useful in other subroutines. The first proposed test added to RosterTest depend upon TestUtil's methods. The second does not.

navijation committed:
TestUtil.java class proposal

This is a static utilities class I made (primarily) for the purposes of test-case generating functions and (possibly) other helper methods that aren't specific to any class. The methods currently available can create a fake NRList and fake PeerInteractions. These can generate very large test cases (albeit ones that conform to a prescribed model).

I directly checked out the adding-tests branch and it compiles under Eclipse even with this file added. I'm not really sure how to test if it compiles under Travis, though.

navijation committed:
Added randomized tests to RosterTest

The second test is deterministic. The first one is randomized, using TestUtil's methods to create fake Roster information and fake PeerInteractions. It might make sense to add print statements to reveal the details of failing cases, though, since they are generated at random.

navijation committed:
Create TestUtil.java

This is a static utilities class I made (primarily) for the purposes of test-case generating functions and (possibly) other helper methods that aren't specific to any class. The methods currently available can create a fake NRList and fake PeerInteractions. These can generate very large test cases (albeit ones that conform to a prescribed model).

I directly checked out the adding-tests branch and it compiles under Eclipse even with this file added. I'm not really sure how to test if it compiles under Travis, though.

Togira commented on TestUtil.java:
```diff
@@ -0,0 +1,270 @@
+					output.add(wordExtractor.next().replace("\"", "\"\""));
+				wordExtractor.close();
+			}
+		output.sort(naturalOrder);
```
The sort function is undefined causing Travis to fail.

navijation replied:
I think this is because Travis uses Java SE 7? I'll change my local compiler and fix this. I would recommend compiling the project Java SE 8 though. It's a large improvement over SE 7 in terms of features (eg lambdas).

Togira replied:
My machine thinks the method is undefined as well.  It could be a Java version issue but for now, is there an alternative solution so that this line won't cause issues on everyone's computer who is still building against SE 7?

navijation committed:
Conformed to Travis (hopefully)

Hopefully this conforms to Travis CI. I replaced ArrayList.sort(Comparator) [A Java SE 8 feature] with Collections.sort(ArrayList) [A Java SE 7 feature at least, most likely much older].

navijation committed:
Replaced ArrayList.sort(Comparator) with Collections.sort(ArrayList)

Conforms to older versions of Java.

navijation committed:
Fixed RosterTest.java tests

File was out of date, leading to buggy ArrayIndexOutOfBoundsException(). However, the second test managed to catch an error with the Iterator on my machine, but only some of the time. So I updated the second test to run 100 times, just to be safe (probably not the best practice, but whatever). I'll fix Roster.java shortly.

navijation committed:
Fixed Roster.StudentIterator

Got rid of my one-liner hackery from StudentIterator's methods since it led to utter failure in certain cases, which I do not understand to this day. This should work now...

NavneethJayendran commented:
So apparently I put the latest version of TestUtil.java outside of src... I think I'll call it a night after this debacle. If you could be so kind as to delete that file and port the changes over to the real one, I'm at least 60% sure it'll compile.

Commit:
Removed extra TestUtil file.

Commit:
Updated TestUtil name to Utilities to hopefully avoid travis issues without needing to customize the build.xml file.

Togira commented:
This is good to go on my end. I'll leave it up to @pranaygp or someone else to hit merge.

pranaygp commented:
I'm on my phone right now so I can't hit merge. I accept this though, so feel free to merge it @Togira
